contracts/bonding/BondingManager.sol

```diff
@@ -137,23 +153,33 @@ contract BondingManager is IBondingManager, Manager {
     function transcoder(uint8 _blockRewardCut, uint8 _feeShare, uint256 _pricePerSegment)
         external
         whenSystemNotPaused
+        currentRoundInitialized
         returns (bool)
     {
-        // Current round must be initialized
-        require(roundsManager().currentRoundInitialized());
         // Block reward cut must a valid percentage
         require(_blockRewardCut <= 100);
         // Fee share must be a valid percentage
         require(_feeShare <= 100);
+        // Sender must not be a resigned transcoder
+        require(transcoderStatus(msg.sender) != TranscoderStatus.Resigned);
 
         Transcoder storage t = transcoders[msg.sender];
-        t.transcoderAddress = msg.sender;
         t.pendingBlockRewardCut = _blockRewardCut;
         t.pendingFeeShare = _feeShare;
         t.pendingPricePerSegment = _pricePerSegment;
 
-        if (!transcoderPools.isInPools(msg.sender)) {
-            transcoderPools.addTranscoder(msg.sender, 0);
+        if (transcoderStatus(msg.sender) == TranscoderStatus.NotRegistered) {
+            t.delegatorWithdrawRound = 0;
+
+            if (delegatorStatus(msg.sender) == DelegatorStatus.Bonded && delegators[msg.sender].delegationTarget == msg.sender) {
+                // Sender is delegated to itself
+                // Add new transcoder with delegated stake to pools. Fails for insufficient delegated stake
+                transcoderPools.addTranscoder(msg.sender, delegators[msg.sender].bondedAmount);
+            } else {
+                // Sender is not delegated to itself
+                // Add new transcoder with zero stake to pools. Fails if pools are full
+                transcoderPools.addTranscoder(msg.sender, 0);
```

Comment:
Is there any chance that someone else has delegated towards this user, and therefore their total stake would not be == 0?

Reply:
Yeah I wasn't sure how to handle this case without including more on-chain state since users currently do not know how much others have delegated to them unless they are already registered transcoders (at that point `transcoderPools.transcoderStake()` can provide their total stake). I'll need to think about this one

Reply:
One option is to include a `delegatedAmount` field for delegators. Whenever a delegator delegates to a delegatee, the delegatee's `delegatedAmount` field increases by the amount of tokens delegated.